python/cudaq/__init__.py

```diff
@@ -30,7 +30,8 @@
             os.environ["CUDAQ_DYNLIBS"] += f":{cudart_path}"
     except:
         import importlib.util
-        if not importlib.util.find_spec("cuda-quantum") is None:
+        package_spec = importlib.util.find_spec("cuda-quantum")
+        if not package_spec is None and not package_spec.loader is None:
```

Comment:
Is this correct? It seems like it should be
```suggestion
        if package_spec is None or package_spec.loader is None:
```

Reply:
The idea here is that the warning should only be printed if the wheel was installed with pip, since only in this case we actually expect to pick up the cuquantum dependencies as pip packages as well. If the package was installed via pip, then the package_spec.loader has a value, i.e. is not None. What I hadn't realized, or maybe it has changed, is that the package_spec itself potentially still has a value even when cudaq was not pip installed but instead built with the build_cudaq script. In that case we don't want to give this warning, since it will instead pick up the cuquantum dependencies from where they were installed/configured during the build. So I think while it looks a bit cryptic, given the warning what package spec is not None *and* the loader is not None, is indeed the correct behavior.